pkg/bosh/manifest/dns.go

```diff
@@ -0,0 +1,323 @@
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, obj := range []v12.Object{&configMap, &deployment, &service} {
+		err := setOwner(obj)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, c, &configMap, configMapMutateFn(&configMap))
+	if err != nil {
+		return err
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, c, &deployment, deploymentMapMutateFn(&deployment))
+	if err != nil {
+		return err
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, c, &service, mutate.ServiceMutateFn(&service))
+	if err != nil {
+		return err
+	}
+
+	dns.localDNSIP = service.Spec.ClusterIP
+	return err
+}
+
+var re = regexp.MustCompile(`nameserver\s+([\d\.]*)`)
+
+func (dns *boshDomainNameService) rootDNSIP() string {
+	content, err := ioutil.ReadFile("/etc/resolv.conf")
+	if err != nil {
+		content = []byte{}
+	}
+	return GetNameserverFromResolveConfig(content)
+}
+
+// GetNameserverFromResolveConfig read nameserver from resolve.conf
+func GetNameserverFromResolveConfig(content []byte) string {
+	found := re.FindSubmatch(content)
+	if len(found) == 0 {
+		ctxlog.Errorf(context.TODO(), "No nameserver found in resolve.conf using 1.1.1.1")
+		return "1.1.1.1"
+	}
+	return string(found[1])
+
+}
+
+type simpleDomainNameService struct {
+}
+
+// FindServiceNames see interface
+func (dns *simpleDomainNameService) FindServiceNames(instanceGroupName string, deploymentName string) []string {
+	return []string{serviceName(instanceGroupName, deploymentName, 63)}
+}
+
+// HeadlessServiceName see interface
+func (dns *simpleDomainNameService) HeadlessServiceName(instanceGroupName string, deploymentName string) string {
+	return serviceName(instanceGroupName, deploymentName, 63)
+}
+
+// ConfigurePod see interface
+func (dns *simpleDomainNameService) ConfigurePod(podSpec *v1.PodSpec) {
+}
+
+// Reconcile see interface
+func (dns *simpleDomainNameService) Reconcile(ctx context.Context, c client.Client, setOwner func(object v12.Object) error) error {
+	return nil
+}
+
+func serviceName(instanceGroupName string, deploymentName string, maxLength int) string {
+	serviceName := fmt.Sprintf("%s-%s", deploymentName, names.Sanitize(instanceGroupName))
+	if len(serviceName) > maxLength {
+		sumHex := md5.Sum([]byte(serviceName))
+		sum := hex.EncodeToString(sumHex[:])
+		serviceName = fmt.Sprintf("%s-%s", serviceName[:maxLength-len(sum)-1], sum)
+	}
+	return serviceName
+}
+
+func configMapMutateFn(configMap *v1.ConfigMap) controllerutil.MutateFn {
+	updated := configMap.DeepCopy()
+	return func() error {
+		configMap.Labels = updated.Labels
+		configMap.Annotations = updated.Annotations
+		configMap.Data = updated.Data
+		return nil
+	}
+}
+
+func deploymentMapMutateFn(deployment *v13.Deployment) controllerutil.MutateFn {
+	updated := deployment.DeepCopy()
+	return func() error {
+		deployment.Labels = updated.Labels
+		deployment.Annotations = updated.Annotations
+		deployment.Spec = updated.Spec
+		return nil
+	}
+}
+
+const corefile = `
```

Comment:
How about we use `text/template` here to make it more robust?

Reply:
Changing it to `text/template` was a bit overkill, but we refactored it, so  that it is hopefully less errorprone.